src/components/Dashboard/UserBox.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import classnames from 'classnames';
+import { useRouter } from 'next/router';
+
+import { useUserName } from '../AuthenticationProvider';
+import { useAuthentication } from '../AuthenticationProvider';
+import { Row } from '../Row/Row';
+import { Button } from '../Button/Button';
+import style from './UserBox.module.scss';
+
+export const UserBox = () => {
+  const [userBoxVisible, setUserBoxVisible] = useState(false);
+
+  const userName = useUserName();
+  const router = useRouter();
+  const { logout } = useAuthentication();
```

Comment:
If you already have the authentication hook here, you can take the userName out of there 

```suggestion
  const router = useRouter();
  const { logout, userName } = useAuthentication();
```